### ISSUE TITLE
Default application credential check for App Engine seems to be incomplete

### ISSUE THREAD

**Issue by laixer** [google/google-api-java-client#1001]
This is the current check:

``` java
    try {
      systemPropertyClass = forName("com.google.appengine.api.utils.SystemProperty");
    } catch (ClassNotFoundException expected) {
      // SystemProperty will always be present on App Engine.
      return false;
    }
```

If you deploy the "Hello World" application (https://cloud.google.com/appengine/docs/java/) you will see that this check fails.

It's necessary to add a dependency on appengine-api-1.0-sdk for this check to pass.

Is there a better way to detect if the client is running on App Engine?

**Comment by mattwhisenhunt** [comment]
This may have been wrong issue tracker to post your question in. A quick Google search yielded the following result: https://stackoverflow.com/questions/1993199/determine-appengine-for-java-environment-programmatically

**Comment by laixer** [comment]
Hi Matt,

I work on GCP... This is definitely the right repo and the issue is still there. 
The default credentials check for App Engine is based on a system property that is *not* always present.

**Comment by mattwhisenhunt** [comment]
Fixed in #1089
